### PR TITLE
Add dry_run configuration parameter

### PR DESCRIPTION
**Pull request by asmorodskyi**
This will allow us to test new functionality without
fear to break something in real world. When this parameter
is True every API call to all public cloud providers is replaced
with log message

### CODE REVIEW

**Comment by pdostal** [comment]
This looks very good!

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    with open(filename, "w") as f:
+        f.write(add)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def generalSetup():
+    tmpFile = tempfile.mkstemp()
+    webui.settings.CONFIG_FILE = tmpFile[1]
+    set_pcw_ini(tmpFile[1])
+    yield tmpFile[1]
```
Comment (cfconrad) [inline tests/conftest.py:17]: not sure about that yield here, can you explain why not `return` or yield in a loop?

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import tempfile
+
+
+def set_pcw_ini(filename, add=''):
+    with open(filename, "w") as f:
+        f.write(add)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def generalSetup():
```
Comment (asmorodskyi) [inline tests/conftest.py:13]:
```suggestion
def pcw_ini():
```

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    tmpFile = tempfile.mkstemp()
+    webui.settings.CONFIG_FILE = tmpFile[1]
+    set_pcw_ini(tmpFile[1])
+    yield tmpFile[1]
+    # to make sure that we have clean file before next test
+    set_pcw_ini(tmpFile[1])
```
Comment (asmorodskyi) [inline tests/conftest.py:19]: we don't need this

Comment (asmorodskyi) [reply tests/conftest.py:19]: but we need to cleanup a file

**Review by cfconrad — APPROVED** [review]
LGTM

**Pull request merged**